File: selfstudy/js/fishes_sum.js
/*
Task: Fishes - Sum
Calculate the total sum of fish lengths.
*/

function sum_fishes(fishes) {
  return fishes.reduce((a, b) => a + b, 0);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: 60 },
  { input: [5, 5, 5], expected: 15 },
  { input: [1, 2], expected: 3 },
  { input: [100], expected: 100 },
  { input: [], expected: 0 }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('sum_fishes', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(sum_fishes(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, sum_fishes(t.input) === t.expected ? "PASS" : "FAIL");
  });
}

File: selfstudy/golang/fishes_sum.go
/*
Task: Fishes - Sum
Calculate the total sum of fish lengths.
*/

package main

import "fmt"

func SumFishes(fishes []int) int {
	sum := 0
	for _, v := range fishes {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, 60},
		{[]int{5, 5, 5}, 15},
		{[]int{1, 2}, 3},
		{[]int{100}, 100},
		{[]int{}, 0},
	}
	
	for idx, t := range tests {
		res := SumFishes(t.input)
		status := "FAIL"
		if res == t.expected {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}
